### ISSUE TITLE
Verb option help is not rendering Enum value options

### ISSUE THREAD

**Issue by ericnewton76** [commandlineparser/commandline#46]
<a href="https://github.com/kensykora"><img src="https://avatars3.githubusercontent.com/u/64756?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [kensykora](https://github.com/kensykora)**
_Thursday Oct 01, 2015 at 14:32 GMT_
_Originally opened as https://github.com/gsscoder/commandline/issues/246_

----

When I create a verb object with a verb that has an enum value option, the help for the verb is not rendering. See example here:  https://gist.github.com/kensykora/f486cdf239a823390409

Basically, any enum that's a positional parameter doesn't show up in autobuild help. It'd also be nice if the enum displayed it's string value rather than the int value, maybe even used the System.ComponentModel.DescriptionAttribute option?

**Comment by ericnewton76** [comment]
<a href="https://github.com/gsscoder"><img src="https://avatars3.githubusercontent.com/u/1194228?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [gsscoder](https://github.com/gsscoder)**
_Monday Oct 05, 2015 at 22:42 GMT_

----

:paperclip: Please read: #250. Thanks for collaboration and patience! :sweat_smile:
Giacomo
_P.S.: valid only for me, if a main contrib. is available he's obviously free to support devs/users of the project._

**Comment by ericnewton76** [comment]
<a href="https://github.com/blakecausey"><img src="https://avatars3.githubusercontent.com/u/3493765?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [blakecausey](https://github.com/blakecausey)**
_Wednesday Jan 20, 2016 at 20:30 GMT_

----

This error appears to be in the **TryParseHelpVerb** method where the verb is fetched as the first item of the **args** parameter but for "help" the verb should be the second item. e.g. "myapp help commit" where "commit" is the verb.

The following line shows where it fetches the first item of args.

``` csharp
var verb = args.FirstOrDefault();
```

To prove my point, the following "if" statement looks for "help" in args[0] and will return if it does not find it as shown below:

``` csharp
if (string.Compare(args[0], helpInfo.Right.LongName, GetStringComparison(_settings)) == 0)
```

Below is the complete method.

``` csharp
private bool TryParseHelpVerb(string[] args, object options, Pair<MethodInfo, HelpVerbOptionAttribute> helpInfo, OptionMap optionMap)
{
    var helpWriter = _settings.HelpWriter;
    if (helpInfo != null && helpWriter != null)
    {
        if (string.Compare(args[0], helpInfo.Right.LongName, GetStringComparison(_settings)) == 0)
        {
            // User explicitly requested help
            var verb = args.FirstOrDefault();
            if (verb != null)
            {
                var verbOption = optionMap[verb];
                if (verbOption != null)
                {
                    if (verbOption.GetValue(options) == null)
                    {
                        // We need to create an instance also to render help
                        verbOption.CreateInstance(options);
                    }
                }
            }

            DisplayHelpVerbText(options, helpInfo, verb);
            return true;
        }
    }

    return false;
}
```

I'm unable to get the time to test my theory or even come up with a fix but most likely I would do something like:

``` csharp
var verb = args.Skip(1).FirstOrDefault();
```

**Comment by ericnewton76** [comment]
<a href="https://github.com/DanHarman"><img src="https://avatars2.githubusercontent.com/u/1155948?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [DanHarman](https://github.com/DanHarman)**
_Wednesday Mar 09, 2016 at 16:04 GMT_

----

Enum parsing with verbs as positional arguments is also currently case sensitive when I don't believe it should be.

**Comment by ericnewton76** [comment]
<a href="https://github.com/jkodroff"><img src="https://avatars1.githubusercontent.com/u/410614?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jkodroff](https://github.com/jkodroff)**
_Monday Jun 20, 2016 at 13:29 GMT_

----

@DanHarman I agree.  See: https://github.com/gsscoder/commandline/issues/326
